elastic/kibana
# [maps] fix Dashboards with by-value maps are broken when copied to new space

Closes https://github.com/elastic/kibana/issues/125595 and https://github.com/elastic/kibana/issues/124976

#### Changes
* PR adds maps embeddable migration to migrate existing by-value map embeddable in existing dashboards.
* This PR updates maps embeddable factory to register extract and inject functions to properly extract and inject references for by value embeddables in client and server.

#### Test - newly created dashboards
1) Create dashboard with by-value map that contains documents layer. Save
2) From Stack management -> Saved objects page, copy dashboard to space
3) Open dashboard in space and verify map shows data

#### Test - existing by-value dashboards.
1) In Cloud instance that is 7.16, or 7.17.0, or 8.0.0, Create dashboard with by-value map that contains documents layer. Save
2) export dashboard saved objects
3) Import dashboard into Kibana instance that is running this branch
4) Open dashboard and verify map shows data
5) Copy dashboard to space
6) Open dashboard in space and verify map shows data
